state_estimator:
  ros__parameters:
    noise_parameters:
      process_noise:
        linear_acceleration:  # accelerometer's noise density in MTi-1-series datasheet
        - 0.001176798  # m/s^2/√Hz, x
        - 0.001176798  # m/s^2/√Hz, y
        - 0.001176798  # m/s^2/√Hz, z
        angular_velocity:
        - 0.00012217304  # rad/s/√Hz, roll
        - 0.00012217304  # rad/s/√Hz, pitch
        - 0.00012217304  # rad/s/√Hz, yaw
        foot_position:
        - 1.0  # x
        - 1.0  # y
        - 1.0  # z
        accelerometer_bias:
        - 4.25166e-11  # m/s^2/√Hz, x
        - 4.25166e-11  # m/s^2/√Hz, y
        - 5.25778e-11  # m/s^2/√Hz, z
        gyroscope_bias:
        - 1.46670e-12  # rad/s/√Hz, roll
        - 1.46670e-12  # rad/s/√Hz, pitch
        - 1.46670e-12  # rad/s/√Hz, yaw
        foot_slippage:
        - 1.0  # roll
        - 1.0  # pitch
        - 1.0  # yaw
      observation_noise:
        foot_position:
        - 1e-1  # x
        - 1e-1  # y
        - 1e-1  # z
        foot_slippage:
        - 1e-2  # roll
        - 1e-2  # pitch
        - 1e-2  # yaw
        joint_position:
        - 1e-3  # left_hip_aa
        - 1e-3  # left_hip_fe
        - 1e-3  # left_knee
        - 1e-3  # left_ankle_dpf
        - 1e-3  # left_ankle_ie
        - 1e-3  # right_hip_aa
        - 1e-3  # right_hip_fe
        - 1e-3  # right_knee
        - 1e-3  # right_ankle_dpf
        - 1e-3  # right_ankle_ie
    optimization_parameters:
      # process_noise:
      # - foot_position
      # - foot_slippage
      observation_noise:
      - foot_position
      - foot_slippage
      - joint_position